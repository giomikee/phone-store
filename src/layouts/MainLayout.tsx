import { Outlet } from 'react-router-dom'
import NavBar from '../components/NavBar/NavBar';
import { Box } from '@mui/material';

const MainLayout = () => {
    return (
        <>
            <NavBar />
            <Box sx={{mt:9, mb: 2}}>
                <Outlet />
            </Box>
        </>
    )
}

export default MainLayout