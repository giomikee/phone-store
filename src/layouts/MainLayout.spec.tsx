import { describe, expect, it, vi } from 'vitest';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/vitest';
import { BrowserRouter } from 'react-router-dom';
import MainLayout from './MainLayout';

vi.mock('react-redux', async () => ({
    ...(await vi.importActual('react-redux')),
    useDispatch: vi.fn().mockReturnValue(vi.fn()),
    useSelector: vi.fn().mockReturnValue([]),
}));

vi.mock('../../state/cart/cart.slice');

describe('Given MainLayout component', () => {
    describe('When the component is rendered', () => {
        it('Then it should match the snapshot', () => {
            const { container } = render(<MainLayout />, { wrapper: BrowserRouter });
            expect(container).toMatchSnapshot();
        });
    });
})