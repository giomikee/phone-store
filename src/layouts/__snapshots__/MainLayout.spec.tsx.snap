// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given MainLayout component > When the component is rendered > Then it should match the snapshot 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-zanzjh-MuiPaper-root-MuiAppBar-root"
    style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-5c1adp-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-dense css-10po6k9-MuiToolbar-root"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-1blvszr-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1sh91j5-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="HomeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
            </span>
            Inicio
          </button>
        </a>
        <div
          class="MuiBox-root css-17ls7xt"
        >
          <h1
            class="MuiTypography-root MuiTypography-h4 css-f84ov1-MuiTypography-root"
          >
            Phone Store
          </h1>
        </div>
        <a
          class=""
          href="/cart"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSuccess MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSuccess MuiButton-root MuiButton-text MuiButton-textSuccess MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSuccess css-1ifubkm-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1sh91j5-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="ShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2M1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2"
                />
              </svg>
            </span>
            0
          </button>
        </a>
      </div>
    </div>
  </header>
  <div
    class="MuiBox-root css-r6il6a"
  />
</div>
`;
